xdsl/dialects/builtin.py

```diff
@@ -910,6 +911,19 @@ def from_strings(name: str, value: str,
         return OpaqueAttr([StringAttr(name), StringAttr(value), type])
 
 
+@irdl_op_definition
+class UnrealizedConversionCastOp(Operation):
+    name: str = "builtin.unrealized_conversion_cast"
+
+    input: Annotated[Operand, AnyAttr()]
+    output: Annotated[OpResult, AnyAttr()]
```

Comment:
I think we should use `: Operand` instead, since the `Annotated` does nothing here. Afaik (and @math-fehr can elaborate on this), the `AnyAttr()` constraint is not meant as to be used outside of `irdl` as it is not really a constraint.

Reply:
I don't know, I like using the same format everywhere actually.